Artifacts/windows-certificate-from-keyvault/InstallCertificate.ps1

```diff
@@ -0,0 +1,103 @@
+Param(
+    [ValidateNotNullOrEmpty()]
+    [string]$vaultName,
+    [ValidateNotNullOrEmpty()]
+    [string]$secretName,
+    [ValidateNotNullOrEmpty()]
+    [string]$azureServicePrincipalClientId,
+    [ValidateNotNullOrEmpty()]
+    [string]$azureServicePrincipalKey,
+    [ValidateNotNullOrEmpty()]
+    [string]$azureServicePrincipalTenantId
+)
+
+# Handle all errors in this script.
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this script, unless you want to ignore a specific error.
+    Handle-LastError
+}
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+$ErrorActionPreference = "Stop"
+
+function Handle-LastError
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $message = $error[0].Exception.Message
+    if ($message)
+    {
+        Write-Host -Object "ERROR: $message" -ForegroundColor Red
+    }
+    
+    # IMPORTANT NOTE: Throwing a terminating error (using $ErrorActionPreference = "Stop") still
+    # returns exit code zero from the PowerShell script when using -File. The workaround is to
+    # NOT use -File when calling this script and leverage the try-catch-finally block and return
+    # a non-zero exit code from the catch block.
+    exit -1
+}
+
+
+try{
+
+    if (-not (Get-Module -Name "AzureRm")){
```

Comment:
This test might fail in the latest version of Azure PowerShell. It seems that AzureRM only gets installed if we invoke `Install-Module AzureRM`, but not when we install Azure PowerShell from the WebInstaller (latest I saw is version 4.2). You may want to check it out on a new machine to confirm.